common/src/main/java/owmii/powah/item/WrenchItem.java

```diff
@@ -68,10 +74,44 @@ public InteractionResult onItemUseFirst(ItemStack stack, Level world, BlockPos p
 //                    }
 //                }
             }
+            if (getWrenchMode(stack).rotate()
+                    // Only rotate Powah machines
+                    && state.getBlock() instanceof AbstractEnergyBlock<?,?>) {
```

Comment:
> If we moved the rotation handling into the event handlers I think it would work... and then we could also treat non-Powah wrenches as rotate mode.

What about the wrench mode then?

Reply:
`onItemUseFirst` should be called before the block is used, that's why I expected it to work. On Fabric it's already implemented using an event handler.

Reply:
> The whole thing would be gated by isWrench (the tag check); then have a mode variable initialized with the default for non-Powah wrenches (not sure if it should be rotate or config) that gets reassigned to the wrench's mode if the wrench is a Powah wrench

Where would the mode be? We don't want to be adding NBT to foreign wrench items.

Reply:
> `onItemUseFirst` should be called before the block is used, that's why I expected it to work. On Fabric it's already implemented using an event handler.

Oh, I was testing on Forge just because I had a test world from testing #75, will have another try on Fabric

Reply:
> > The whole thing would be gated by isWrench (the tag check); then have a mode variable initialized with the default for non-Powah wrenches (not sure if it should be rotate or config) that gets reassigned to the wrench's mode if the wrench is a Powah wrench
> 
> Where would the mode be? We don't want to be adding NBT to foreign wrench items.

It wouldn't be on the item; it would just be a global default mode that we fall back to for non-Powah wrenches.
If you want to choose the mode you will still need a Powah wrench.

> `onItemUseFirst` should be called before the block is used, that's why I expected it to work. On Fabric it's already implemented using an event handler.

Tested on Fabric with the orb check added, and it's still not able to rotate, the wrench goes into the orb.

Reply:
Found the problem, energizing orb can be rotated now. Will leave further support for other mods wrenches out of this PR, can be revisited later.